WorkflowUI/Tests/WorkflowUIViewControllerTests.swift

```diff
@@ -0,0 +1,239 @@
+                animated: false,
+                isFirstAppearance: false
+            )
+        )
+    }
+}
+
+// MARK: Known Subclass Tests
+
+final class WorkflowUIViewControllerSubclassesTests: WorkflowUIObservationTestCase {
+    // MARK: Subclass Tests
+
+    func test_describedVC_emitsVCLifecycleEvents() {
+        let subject = DescribedViewController(description: .testing)
+
+        let observedEvents = observationEvents(from: subject) {
+            subject.viewWillAppear(false)
+            subject.viewDidAppear(false)
+
+            subject.viewWillLayoutSubviews()
+            subject.viewDidLayoutSubviews()
+        }
+
+        let expectedEventDescriptors: [EventDescriptor] = [
+            .viewWillAppear,
+            .viewDidAppear,
+            .viewWillLayoutSubviews,
+            .viewDidLayoutSubviews,
+        ]
+
+        XCTAssertEqual(
+            observedEvents.map(\.descriptor),
+            expectedEventDescriptors
+        )
+    }
+
+    func test_screenVC_emitsVCLifecycleEvents() {
+        let subject = ScreenViewController(screen: TestScreen(), environment: .empty)
+
+        let observedEvents = observationEvents(from: subject) {
+            subject.viewWillAppear(false)
+            subject.viewDidAppear(false)
+
+            subject.viewWillLayoutSubviews()
+            subject.viewDidLayoutSubviews()
+        }
+
+        let expectedEventDescriptors: [EventDescriptor] = [
+            .viewWillAppear,
+            .viewDidAppear,
+            .viewWillLayoutSubviews,
+            .viewDidLayoutSubviews,
+        ]
+
+        XCTAssertEqual(
+            observedEvents.map(\.descriptor),
+            expectedEventDescriptors
+        )
+    }
+
+    func test_hostingController_emitsVCLifecycleEvents() {
+        let subject = WorkflowHostingController(workflow: TestWorkflow())
+
+        let observedEvents = observationEvents(from: subject) {
+            subject.viewWillAppear(false)
+            subject.viewDidAppear(false)
+
+            subject.viewWillLayoutSubviews()
+            subject.viewDidLayoutSubviews()
+        }
+
+        let expectedEventDescriptors: [EventDescriptor] = [
+            .viewWillAppear,
+            .viewDidAppear,
+            .viewWillLayoutSubviews,
+            .viewDidLayoutSubviews,
+        ]
+
+        XCTAssertEqual(
+            observedEvents.map(\.descriptor),
+            expectedEventDescriptors
+        )
+    }
+}
+
+private struct TestWorkflow: Workflow {
+    func makeInitialState() -> Int { 0 }
+
+    func render(state: Int, context: RenderContext<TestWorkflow>) -> TestScreen {
+        TestScreen(value: state)
+    }
+}
+
+private struct TestScreen: Screen {
+    var value: Int = 0
+
+    func viewControllerDescription(environment: ViewEnvironment) -> ViewControllerDescription {
+        .testing
+    }
+}
+
+private extension ViewControllerDescription {
+    static var testing: Self {
+        .init(build: { UIViewController() }, update: { _ in })
```

Comment:
I think this would be the only conflict when compared to the ViewEnvironmentUI work—`ViewControllerDescription` requires an `environment` parameter in that feature branch. It should be easy to reconcile depending on who lands their work first!